client/src/commonMain/kotlin/com/algolia/search/model/rule/OptionalFilters.kt

```diff
@@ -0,0 +1,52 @@
+package com.algolia.search.model.rule
+
+import com.algolia.search.exception.AlgoliaClientException
+import com.algolia.search.serialize.internal.isJsonArray
+import com.algolia.search.serialize.internal.isString
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonContentPolymorphicSerializer
+import kotlinx.serialization.json.JsonElement
+
+/**
+ * Filters to promote or demote records in the search results.
+ * Optional filters work like facet filters, but they don't exclude records from the search results.
+ * Records that match the optional filter rank before records that don't match. If you're using a negative filter `facet:-value`, matching records rank after records that don't match.
+ * - Optional filters don't work on virtual replicas.
+ * - Optional filters are applied _after_ `sort-by` attributes.
+ * - Optional filters don't work with numeric attributes.
+ *
+ * Implementations:
+ * - [List<OptionalFilters>] - *[OptionalFilters.of]*
+ * - [String] - *[OptionalFilters.of]*
+ */
+@Serializable(OptionalFiltersSerializer::class)
+public sealed interface OptionalFilters {
+    @Serializable
+    @JvmInline
+    public value class ListOfOptionalFiltersValue(public val value: List<OptionalFilters>) : OptionalFilters
```

Comment:
This feels wrong 😅, shouldn't this be `List<String>`?

Reply:
I think not: the end goal is to make it recursive on itself
It can be either a String or I a list of String or a list of list of String and so on 😁